src/supervisor.adoc

```diff
@@ -824,6 +825,17 @@ The definitions of the CBCFE and CBIE fields will be furnished by the
 forthcoming Zicbom extension. Their allocations within `senvcfg` may
 change prior to the ratification of that extension.
 
+If the Svkt extension is implemented, the SVKT field affects the behavior of
+instruction fetches and explicit memory accesses.
+When SVKT=0, instruction fetches and explicit memory accesses proceed as
+though the Svkt extension were not implemented.
+When SVKT=1, an instruction fetch or explicit memory access with effective
+privilege mode U or VU that attempts to access a virtual address {ge}
+2^SXLEN-1^ raises a page-fault exception corresponding to the original access
+type.
```

Comment:
Do we want more flexibility here? e.g. for 32-bit runtime. AFIAK, 32-bit Linux uses virtual addresses starting from `0xc0000000` as supervisor mapping. Different OSes may define it differently on 32-bit platform though, due to contested address space, e.g. windows fits into this half/half s/u division, IIRC.

Reply:
Two comments here

1) I am not sure, if it's helpful for 32bit. 2 GB  (or 1GB when supervisor_addr_start=`0xc0000000`) is relatively small address space to have any significant benefit from KASLR.
So may be mandating only for 64 bit, probably makes it cleaner. 
Side note: Not that MSFT Windows should be of concern here but Windows 32bit also have an option to split address space in 3GB/1GB. Anyways 32bit windows is deprecated now.

2) `2^SXLEN-1` doesn't work when pointer masking is enabled because canonical checks are relaxed.

Instead of `2^SXLEN-1`, it should be

```
2^VABITS-1
VABITS = 39 if Sv39
VABITS = 48 if Sv48
VABITS = 57 if Sv57

If Bare mode then this bit has no effect?
```

Reply:
Never mind about comment # 2.
Shouldn't be a problem, as long as effective address is calculated after masking is performed and generated address is canonicalized again.

Reply:
@AndybnACT We did discuss that, but the tentative plan is to support Svkt only for RV64 for the reason that @deepak0414 mentioned: KASLR is seemingly much less beneficial on RV32.  But if we did add an RV32 variant, we of course would design it so that it would be usable by the Linux kernel.  This would either mean using a 3G/1G split (most likely) or offering the choice between 3G/1G or 2G/2G (less likely).

@deepak0414 Right, re: # 2, the address transformation is applied before any virtual-address handling, and so there is no problem with pointer masking.

Reply:
I added a note about not having Svkt in RV32. 9a9ba46df4e7bc7f72288c33dc97e020324f59ff